src/opengl_render.c

```diff
@@ -100,37 +105,48 @@ void gl_renderRect( double x, double y, double w, double h, const glColour *c )
  *    @param y Y position to render rectangle at.
  *    @param w Rectangle width.
  *    @param h Rectangle height.
+ *    @param line_width Outline thickness or 0 for a filled rectangle.
+ *    @param rx Corner curve lenght along X axis.
+ *    @param ry Corner curve lenght along Y axis.
  *    @param c Rectangle colour.
  */
-void gl_renderRectEmpty( double x, double y, double w, double h,
+void gl_renderRoundRect( double x, double y, double w, double h,
+                         double line_width, double rx, double ry,
                          const glColour *c )
 {
    mat4 projection = gl_view_matrix;
    mat4_translate_scale_xy( &projection, x, y, w, h );
 
-   gl_renderRectH( &projection, c, 0 );
+   gl_renderRectH( &projection, c, line_width, rx, ry );
 }
 
 /**
  * @brief Renders a rectangle.
  *
  *    @param H Transformation matrix to apply.
- *    @param filled Whether or not to fill.
  *    @param c Rectangle colour.
+ *    @param line_width Thickness of the outline or 0 to fullfill it.
+ *    @param rx Corner curve lenght along X axis.
+ *    @param ry Corner curve lenght along Y axis.
  */
-void gl_renderRectH( const mat4 *H, const glColour *c, int filled )
+void gl_renderRectH( const mat4 *H, const glColour *c, int line_width, int rx,
+                     int ry )
 {
-   gl_beginSolidProgram( *H, c );
-   if ( filled ) {
-      gl_vboActivateAttribOffset( gl_squareVBO, shaders.solid.vertex, 0, 2,
-                                  GL_FLOAT, 0 );
-      glDrawArrays( GL_TRIANGLE_STRIP, 0, 4 );
+   if ( conf.round_gui && rx && ry ) {
```

Comment:
I don't like the conditioning the behaviour here. If you, for example, try to draw a curved rectangle irregardless of the gui setting, it won't work. This is probably more important for the Lua API. Would probably be best to have this set when rendering, for example an if statement in renderPane for the C toolkit, and some equivalent wrapper function for luatk or whatever.